source/test/test_datastore.py

```diff
@@ -0,0 +1,100 @@
+        )
+        self.conn.put_item(
+            TableName = DOCUMENTS_TABLE_NAME,
+            Item={
+                "documentId": {"S" : "b1a54fda-1809-49d7-8f19-0d1688eb65b9"},
+                "objectName": {"S": "public/samples/Misc/expense.png"},
+                "bucketName": {"S": "dusstack-sample-s3-bucket"},
+                "documentStatus": {"S": "IN_PROGRESS"}
+            }
+        )
+        self.conn.put_item(
+            TableName = DOCUMENTS_TABLE_NAME,
+            Item={
+                "documentId": {"S" : "b1a99fda-1809-49d7-8f19-0d1688eb65b9"},
+                "objectName": {"S": "public/samples/Misc/expense.png"},
+                "bucketName": {"S": "dusstack-sample-s3-bucket"},
+                "documentStatus": {"S": "IN_PROGRESS"}
+            }
+        )
+        self.ds = datastore.DocumentStore(DOCUMENTS_TABLE_NAME,OUTPUT_TABLE_NAME)
+    
+    def test_create_document_success(self):
+        bucketName = "dusstack-sample-s3-bucket"
+        objectName = "public/samples/Finance/report.pdf"
+        documentId = "b1a66fda-1809-49d7-8f19-0d1688eb65b9"
+        response = self.ds.createDocument(documentId, bucketName, objectName)
+        self.assertEqual(response, None)
+    
+    def test_create_duplicate_documentid_throws_error(self):
+        bucketName = "dusstack-sample-s3-bucket"
+        objectName = "public/samples/Finance/report.pdf"
+        documentId = "b1a54fda-1809-49d7-8f19-0d1688eb65b9"
+        response = self.ds.createDocument(documentId, bucketName, objectName)
+        self.assertEqual(response, {'Error': 'Document already exist.'})
+    
+    def test_update_document_status_success(self):
+        documentId = "b1a54fda-1809-49d7-8f19-0d1688eb65b9"
+        response = self.ds.updateDocumentStatus(documentId, "FAILED")
+        self.assertEqual(response, None)
+    
+    def test_update_document_status_throws_error_when_document_doesnot_exist(self):
+        documentId = "b1333fda-1809-49d7-8f19-0d1688eb65b9"
+        response = self.ds.updateDocumentStatus(documentId, "FAILED")
+        self.assertEqual(response, {'Error': 'Document does not exist.'})
+    
+    def test_mark_document_complete_success(self):
+        documentId = "b1a54fda-1809-49d7-8f19-0d1688eb65b9"
+        response = self.ds.markDocumentComplete(documentId)
+        documentStatus = self.conn.get_item(
+            Key={'documentId': {'S': documentId}},
+            TableName=DOCUMENTS_TABLE_NAME
+        )['Item']['documentStatus']['S']
+        self.assertEqual(documentStatus, "SUCCEEDED")
+        self.assertEqual(response, None)
+    
+    def test_delete_document_success(self):
+        documentId = "b1a54fda-1809-49d7-8f19-0d1688eb65b9"
+        self.ds.deleteDocument(documentId)
+        response = self.conn.get_item(
+            Key={'documentId': {'S': documentId}},
+            TableName=DOCUMENTS_TABLE_NAME
+        )
+        self.assertTrue('Item' not in response)
+    
```

Comment:
Is there a reason to not test getTable and getDocument functions? Could we add those as well?

Reply:
I missed those. I will add it.